File: silver/10773/c++/main.cpp
#include <iostream>
#include <stack>

int main()
{
    int num, number, count = 0;
    std::stack<int> array;
    std::cin >> num;

    for (int i = 0; i < num; i++) {
        std::cin >> number;
        if (number == 0) { array.pop();
        } else { array.push(number); }
    }
    while (array.size()) {
        count += array.top();
        array.pop();
    }
    std::cout << count;
    return 0;
}

File: silver/10773/python/main.py

if (__name__ == "__main__"):
    num = int(input())
    stack = list()
    for i in range(num):
        data = int(input())
        if (data == 0):
            stack.pop(-1)
        else:
            stack.append(data)
    print(sum(stack))